kentik_api_library/kentik_api/synthetics/api_transport_grpc.py

```diff
@@ -0,0 +1,880 @@
+        set_status_req = SetTestStatusRequest(id=id, status=pb_status)
+        self._admin.SetTestStatus(request=set_status_req, metadata=self._credentials, target=self._url)
+
+    def agent_list(self, **kwargs) -> List[Agent]:
+        pb_agents = self._admin.ListAgents(ListAgentsRequest(), metadata=self._credentials, target=self._url).agents
+        return [pb_to_agent(agent) for agent in pb_agents]
+
+    def agent_get(self, **kwargs) -> Agent:
+        id = str(kwargs["id"])
+        response = self._admin.GetAgent(GetAgentRequest(id=id), metadata=self._credentials, target=self._url).agent
+        return pb_to_agent(response)
+
+    def agent_patch(self, **kwargs) -> Agent:
+        pb_agent = pb_from_agent(kwargs["agent"])
+        pb_agent.name = ""  # AgentPatch doesn't accept name
+        mask = FieldMask(paths=[kwargs["mask"]])
+        patch_agent_req = PatchAgentRequest(agent=pb_agent, mask=mask)
+        response = self._admin.PatchAgent(request=patch_agent_req, metadata=self._credentials, target=self._url)
+        return pb_to_agent(response.agent)
+
+    def agent_delete(self, **kwargs) -> None:
+        id = str(kwargs["id"])
+        self._admin.DeleteAgent(DeleteAgentRequest(id=id), metadata=self._credentials, target=self._url)
+
+    def get_health_for_tests(self, **kwargs) -> List[TestHealth]:
+        ids = [str(id) for id in kwargs["test_ids"]]
+        agents = [str(id) for id in kwargs["agent_ids"]]
+        tasks = [str(id) for id in kwargs["task_ids"]]
+        start = Timestamp(seconds=int(kwargs["start_time"].timestamp()))
+        end = Timestamp(seconds=int(kwargs["end_time"].timestamp()))
+        get_health_req = GetHealthForTestsRequest(
+            ids=ids,
+            start_time=start,
+            end_time=end,
+            agent_ids=agents,
+            task_ids=tasks,
+            augment=kwargs["augment"],
+        )
+        response = self._data.GetHealthForTests(request=get_health_req, metadata=self._credentials, target=self._url)
+        return pb_to_health_tests(response.health)
+
+    def get_trace_for_test(self, **kwargs) -> GetTraceForTestResponse:
+        id = str(kwargs["id"])
+        agents = [str(id) for id in kwargs["agent_ids"]]
+        ips = [str(ip) for ip in kwargs["target_ips"]]
+        start = Timestamp(seconds=int(kwargs["start_time"].timestamp()))
+        end = Timestamp(seconds=int(kwargs["end_time"].timestamp()))
+        get_trace_req = GetTraceForTestRequest(
+            id=id,
+            start_time=start,
+            end_time=end,
+            agent_ids=agents,
+            target_ips=ips,
+        )
+        response = self._data.GetTraceForTest(request=get_trace_req, metadata=self._credentials, target=self._url)
+        return pb_to_trace_response(response)
+
+
+def populate_test_from_pb(v: pbTest, out: SynTest) -> None:
+    out.name = v.name
+    out.type = TestType(v.type)
+    out.status = PB_TO_TEST_STATUS[v.status]
+    out.deviceId = ID(v.device_id)
```

Comment:
No need to bother with `deviceId`. It is internal and the API ignores it.

Reply:
removed